pkg/middleware/auth_proxy.go

```diff
@@ -36,46 +31,15 @@ func initContextWithAuthProxy(ctx *m.ReqContext, orgID int64) bool {
 		return true
 	}
 
-	// initialize session
-	if err := ctx.Session.Start(ctx.Context); err != nil {
-		log.Error(3, "Failed to start session. error %v", err)
-		return false
-	}
-
-	defer func() {
-		if err := ctx.Session.Release(); err != nil {
-			ctx.Logger.Error("failed to save session data", "error", err)
-		}
-	}()
-
 	query := &m.GetSignedInUserQuery{OrgId: orgID}
+	userID, err := store.Get(proxyHeaderValue)
```

Comment:
Two different cache keys are being used. We should generate it here and re-use below `key := fmt.Sprintf("auth-proxy-sync-ttl:%s", proxyHeaderValue)`